source/opt/ccp_pass.cpp

```diff
@@ -258,7 +259,19 @@ bool CCPPass::PropagateConstants(ir::Function* fp) {
 
   propagator_ =
       std::unique_ptr<SSAPropagator>(new SSAPropagator(context(), visit_fn));
-  if (propagator_->Run(fp)) {
+  bool modified = propagator_->Run(fp);
+
+#ifndef NDEBUG
+  // Verify all visited values have settled. No value that has been simulated
+  // should end on not interesting.
+  fp->ForEachInst([this](ir::Instruction* inst) {
+    assert(!propagator_->HasStatus(inst) ||
+           propagator_->Status(inst) != SSAPropagator::kNotInteresting &&
+               "Unsettled value");
+  });
+#endif
```

Comment:
Not that is has to be done, but could we make this check part of the generic propagator.  That way we automatically get the checks when someone else tried to use it.

Reply:
Made this change when fixing the compile error.